project/auth.py

```diff
@@ -162,16 +170,73 @@ def pronos():
+    # in the database (gameid, score1, score2, bo and datetime)
+    pronos_teams = list(common_entries(pronos_team1, pronos_team2,pronos_bo, heure_pronos))
+
+    for prono in pronos_teams:
+        # verification of the score and time of the bet
+        if ((int(prono[1]) + int(prono[2]) <= int(prono[3])) and
+            ((int(prono[1]) == (int(prono[3])//2 + 1)) or (int(prono[2]) == (int(prono[3])//2 + 1))) and
+            (datetime.strptime(heure_prono, '%Y-%m-%d %H:%M:%S') < datetime.strptime(prono[4], '%Y-%m-%d %H:%M:%S'))
+        ):
+            # check if there is an existing prediction for this user and this game
+            row = GameProno.query.filter_by(userid=current_user.id, gameid=prono[0]).first()
+
+            if row:
+                # Update existing prediction
+                db.session.execute(
+                    update(GameProno)
+                    .where(GameProno.userid == current_user.id)
+                    .where(GameProno.gameid == prono[0])
+                    .values(prono_team_1=int(prono[1]), prono_team_2=int(prono[2]))
+                )
+            else:
+                # Add new prediction
+                new_prono = GameProno(userid=current_user.id, gameid=prono[0], prono_team_1=int(prono[1]),
+                                      prono_team_2=int(prono[2]))
+                db.session.add(new_prono)
+            db.session.commit()
+
+
+    current_user_league_list = get_current_user_league_list()
+    return render_template('pronos.html', league_list=current_user_league_list, leagueid=0)
+
 
 @auth.route('/pronos_show_league/<leaguename>', methods=['POST'])
 @login_required
 def pronos_show_league(leaguename):
     leagueid = get_leagueid_from_leaguename(leaguename)
     current_user_league_list = get_current_user_league_list()
-    games = Game.query.filter_by(leagueid=leagueid).order_by(Game.game_datetime).all()
-    records = [u.__dict__ for u in games]
-    for r in records:
-        r.pop('_sa_instance_state')
+    query = (select(User.id.label("userid")
+                    , Game.score_team_1, Game.score_team_2, Game.leagueid, Game.bo
+                    , Game.game_datetime, Game.team_1, Game.team_2, Game.id.label("gameid")
+                    , GameProno.prono_team_1, GameProno.prono_team_2
+                    )
+             .join(UserLeague, UserLeague.userid == User.id)
+             .join(Game, UserLeague.leagueid == Game.leagueid)
+             .join(GameProno, and_(GameProno.userid == User.id, GameProno.gameid == Game.id), isouter=True)
+             .where(Game.leagueid == leagueid)
+             .where(current_user.id == User.id)
+             .order_by(Game.game_datetime)
+             )
+    pronos = db.session.execute(query).all()
+    records=[]
+    if pronos:
+        pronos = pd.DataFrame(pronos)
+        columns_to_integer = ['score_team_1', 'score_team_2', 'prono_team_1', 'prono_team_2']
```

Comment:
Ajouter l'information "est-ce que le pronostic est encore modifiable ?" , pour différencier l'affichage.
```suggestion
        pronos['editable'] = datetime.now() < pronos["game_datetime"]
        columns_to_integer = ['score_team_1', 'score_team_2', 'prono_team_1', 'prono_team_2']
```

Reply:
Ici ça te renvoie un True or False sans methode straptime ?